programs/aes/aescrypt2.c

```diff
@@ -105,6 +108,8 @@ int main( int argc, char *argv[] )
     {
         mbedtls_printf( "  ! mbedtls_md_setup() returned -0x%04x\n", -ret );
         goto exit;
+    } else {
```

Comment:
This else and resetting `ret` would not be necessary if you used a separate variable for the exit code and for the return value from `mbedtls_md_setup()`.